src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1663,27 +1664,34 @@ public SolutionState WithFrozenPartialCompilationIncludingSpecificDocument(Docum
                         _latestSolutionWithPartialCompilation.TryGetTarget(out currentPartialSolution);
                     }
 
-                    // if we don't have one or it is stale, create a new partial solution
-                    if (currentPartialSolution == null
-                        || (DateTime.UtcNow - _timeOfLatestSolutionWithPartialCompilation).TotalSeconds >= 0.1
-                        || _documentIdOfLatestSolutionWithPartialCompilation != documentId)
+                    var reuseExistingPartialSolution =
+                        currentPartialSolution != null &&
+                        (DateTime.UtcNow - _timeOfLatestSolutionWithPartialCompilation).TotalSeconds < 0.1 &&
```

Comment:
what's the purpose of the time check?

Reply:
I believe the code clean up I did make control flow simpler since it made removed all nested branching.